pkg/csi/node/server.go

```diff
@@ -182,8 +184,59 @@ func (server *Server) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGetVo
 	}, nil
 }
 
-// NodeExpandVolume returns unimplemented error.
+// NodeExpandVolume handles expand volume request.
 // reference: https://github.com/container-storage-interface/spec/blob/master/spec.md#nodeexpandvolume
-func (server *Server) NodeExpandVolume(ctx context.Context, in *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "unimplemented")
+func (server *Server) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
+	klog.V(3).InfoS("Expand volume requested",
+		"volumeID", req.GetVolumeId(),
+		"VolumePath", req.GetVolumePath())
+
+	volumeID := req.GetVolumeId()
+	if volumeID == "" {
+		return nil, status.Error(codes.InvalidArgument, "volume ID missing in request")
+	}
+	if req.GetVolumePath() == "" {
+		return nil, status.Error(codes.InvalidArgument, "volumePath missing in request")
+	}
+
+	volume, err := client.VolumeClient().Get(ctx, volumeID, metav1.GetOptions{
+		TypeMeta: types.NewVolumeTypeMeta(),
+	})
+	if err != nil {
+		return nil, status.Error(codes.NotFound, err.Error())
```

Comment:
can we send `NotFound` only if the directpvvolume is not found? For other cases, we can return `internal error` code?

Reply:
Done